atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueue.java

```diff
@@ -67,6 +68,15 @@ public static SweepQueue create(TargetedSweepMetrics metrics, KeyValueService kv
         return new SweepQueue(cells, timestamps, progress, deleter, cleaner, shards, metrics);
     }
 
+    /**
+     * Creates an instance of SweepQueue that should be used only for writing to the sweep queue and does not implement
+     * the deleting logic.
+     */
+    public static MultiTableSweepQueueWriter createWriter(TargetedSweepMetrics metrics, KeyValueService kvs,
+            Supplier<Integer> shardsConfig) {
+        return create(metrics, kvs, shardsConfig, new TargetedSweepFollower(ImmutableList.of(), null));
```

Comment:
This is safe (because the TM in the follower should only be invoked if a cleanup task is actually invoked, and there are none), though would be nice if we could make this clearer.

Reply:
Cleaned it up, lmk what you think